snapcraft/plugins/nodejs.py

```diff
@@ -96,6 +96,8 @@ def pull(self):
         super().pull()
         os.makedirs(self._npm_dir, exist_ok=True)
         self._nodejs_tar.download()
+        # we build in 'pull' phase to download all dependencies
+        self.build()
```

Comment:
This for sure solves the problem, but I'm concerned because I doubt that build was designed to be called from pull. There could be many unintended consequences, and the clean_pull is now a little worse for sure. We now can remove the build completely, and do just pull. That sounds weird. I'm not sure, this is a huge thing for a friday evening :)